File: ui/src/types/internal-property.ts
export type InternalProperty = { 
    fullAddress: string; 
    lotPlan?: { 
        lot?: string; 
        plan?: string 
    }; 
    volumeFolio: { 
        volume: string | null; 
        folio: string | null 
    }; 
    status: "KnownVolFol" | "UnknownVolFol"; 
    sourceTrace: { 
        provider?: string; 
        requestId?: string; 
        receivedAt?: string 
    }; 
};

export type InternalPropertyState = {
    propertyId: number,
    data?: InternalProperty
}

File: api/App/Models/InternalProperty.cs
namespace api.Models;

using System.Text.Json.Serialization;

public record InternalProperty( 
    string FullAddress, 
    LotPlan? LotPlan,
    VolumeFolio VolumeFolio,
    VolumeFolioStatus Status,
    SourceTrace SourceTrace
);

public record VolumeFolio(
    string? Volume,
    string? Folio
);

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum VolumeFolioStatus
{
    KnownVolFol,
    UnknownVolFol
}

public record SourceTrace(
    string? Provider,
    string? RequestId,
    DateTimeOffset? ReceivedAt
);
